crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -138,9 +150,224 @@ fn faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() -> anyhow::Res
     Ok(())
 }
 
+// TESTS FOR NEW FAUCET EXECUTION ENVIRONMENT
+// ================================================================================================
+
+#[test]
+fn prove_new_faucet_execution_mint_fungible_asset_succeeds() -> anyhow::Result<()> {
+    // CONSTRUCT AND EXECUTE TX using new faucet that exercises the storage offset validation
+    // --------------------------------------------------------------------------------------------
+    let mut builder = MockChain::builder();
+    let faucet = builder.add_existing_faucet(Auth::BasicAuth, "TST", 200, None)?;
+    let mock_chain = builder.build()?;
+
+    let recipient = Word::from([0, 1, 2, 3u32]);
+    let tag = NoteTag::for_local_use_case(0, 0).unwrap();
+    let aux = Felt::new(27);
+    let note_execution_hint = NoteExecutionHint::on_block_slot(5, 6, 7);
+    let note_type = NoteType::Private;
+    let amount = Felt::new(100);
+
+    tag.validate(note_type).expect("note tag should support private notes");
+
+    let tx_script_code = format!(
+        "
+            begin
+                # pad the stack before call
+                push.0.0.0 padw
+
+                push.{recipient}
+                push.{note_execution_hint}
+                push.{note_type}
+                push.{aux}
+                push.{tag}
+                push.{amount}
+                # => [amount, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
+
+                call.::miden::contracts::faucets::basic_fungible::distribute
+                # => [note_idx, pad(15)]
+
+                # truncate the stack
+                dropw dropw dropw dropw
+            end
+            ",
+        note_type = note_type as u8,
+        recipient = recipient,
+        aux = aux,
+        tag = u32::from(tag),
+        note_execution_hint = Felt::from(note_execution_hint)
+    );
+
+    let tx_script = ScriptBuilder::default().compile_tx_script(tx_script_code)?;
+    let tx_context = mock_chain
+        .build_tx_context(faucet.id(), &[], &[])?
+        .tx_script(tx_script)
+        .build()?;
+
+    let executed_transaction = tx_context.execute_blocking()?;
+
+    prove_and_verify_transaction(executed_transaction.clone())?;
+
+    let fungible_asset: Asset = FungibleAsset::new(faucet.id(), amount.into())?.into();
+
+    let output_note = executed_transaction.output_notes().get_note(0).clone();
+
+    let assets = NoteAssets::new(vec![fungible_asset])?;
+    let id = NoteId::new(recipient, assets.commitment());
+
+    assert_eq!(output_note.id(), id);
+    assert_eq!(
+        output_note.metadata(),
+        &NoteMetadata::new(faucet.id(), NoteType::Private, tag, note_execution_hint, aux)?
+    );
+
+    Ok(())
+}
+
+// TEST FOR create_basic_fungible_faucet FUNCTION
+// ================================================================================================
+
+#[test]
+fn create_basic_fungible_faucet_works() -> anyhow::Result<()> {
```

Comment:
I think I would merge this test into the `miden_lib::account::faucets::tests::faucet_contract_creation` test, since it looks fairly similar. Since this is a more unit-style test that doesn't require a `MockChain` or `TransactionContext` (which are defined in `miden-testing`), I'd move it closer to the function it is testing.